ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -0,0 +1,181 @@
+package com.microsoft.azure.kusto.ingest;
+
+import com.microsoft.azure.kusto.data.*;
+import com.microsoft.azure.kusto.data.exceptions.DataClientException;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException;
+import com.microsoft.azure.kusto.ingest.result.*;
+import com.microsoft.azure.kusto.ingest.source.*;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.*;
+import java.lang.invoke.MethodHandles;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.zip.GZIPOutputStream;
+
+public class StreamingIngestClient implements IngestClient {
+
+    private final static Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private AzureStorageClient azureStorageClient;
+    private Client client;
+    private List<String> mappingRequiredFormats = Arrays.asList("json", "singlejson", "avro");
+
+    StreamingIngestClient(ConnectionStringBuilder csb) throws URISyntaxException {
+        log.info("Creating a new IngestClient");
+        this.client = ClientFactory.createClient(csb);
+        this.azureStorageClient = new AzureStorageClient();
+    }
+
+    @Override
+    public IngestionResult ingestFromFile(FileSourceInfo fileSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        Ensure.argIsNotNull(fileSourceInfo, "fileSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        fileSourceInfo.validate();
+        ingestionProperties.validate();
+
+        try{
+            String filePath = fileSourceInfo.getFilePath();
+            InputStream stream = new FileInputStream(filePath);
+            StreamSourceInfo streamSourceInfo = new StreamSourceInfo(stream,false, fileSourceInfo.getSourceId());
+            streamSourceInfo.setIsCompressed(this.azureStorageClient.isCompressed(filePath));
```

Comment:
is this how we check compression in .NET?

Reply:
.Net doesn't support ingestion of compressed data right now, there is no check and we always compress the data before ingestion